### PR TITLE
Enable having multiple participants

### PR DESCRIPTION
**Pull request by gaborh-da**
This commit enables the application to be started with a subset of all the parties. This way, it can support connecting to multiple participants.

### CODE REVIEW

File: src/main/java/com/digitalasset/refapps/bondissuance/Main.java [hunk 1]
```diff
@@ -60,196 +61,223 @@ public static void main(String[] args) {
...
-          auctionFinalizeBot::calculateCommands,
-          auctionFinalizeBot::getContractInfo);
-
-      Bot.wire(
-          APPLICATION_ID,
-          client,
-          commissionBot.transactionFilter,
-          commissionBot::calculateCommands,
-          commissionBot::getContractInfo);
+      if (parties.hasBank1()) {
```
Comment (tamaskalcza-da) [inline src/main/java/com/digitalasset/refapps/bondissuance/Main.java]: Minor: you consider using `Optional` and `ifPresent`, but I'm fine with this as well.

File: src/test/java/com/digitalasset/refapps/bondissuance/BondIssuanceIT.java [hunk 2]
```diff
@@ -219,10 +227,10 @@ public void testFullWorkflow() throws InvalidProtocolBufferException {
...
             AssetDeposit::fromValue,
             false,
-            assetDepositFinal ->
-                assetDepositFinal.asset.quantity.compareTo(BigDecimal.valueOf(50000000L)) == 0,
-            assetDepositFinal ->
-                assetDepositFinal.asset.quantity.compareTo(BigDecimal.valueOf(8400000L)) == 0));
+            assetDepositFinal -> // 50 000 000 - (42 * 20 000)
+            assetDepositFinal.asset.quantity.compareTo(BigDecimal.valueOf(41600000L)) == 0,
+            assetDepositFinal -> // Redemption value (with coupon): 40.1 * 200 000 * 1.1
+            assetDepositFinal.asset.quantity.compareTo(BigDecimal.valueOf(8822000L)) == 0));
```
Comment (tamaskalcza-da) [inline src/test/java/com/digitalasset/refapps/bondissuance/BondIssuanceIT.java]: Why are these necessary?

File: src/main/java/com/digitalasset/refapps/bondissuance/Main.java (hunk 1 above)
Comment (gaborh-da) [reply src/main/java/com/digitalasset/refapps/bondissuance/Main.java]: First, I had that, but I did not like it because I just needed "a flag of presence", did not need the value itself.

File: src/test/java/com/digitalasset/refapps/bondissuance/BondIssuanceIT.java (hunk 2 above)
Comment (gaborh-da) [reply src/test/java/com/digitalasset/refapps/bondissuance/BondIssuanceIT.java]: The comments or the checks?
The comments, because it took half an hour for us with Gergely to calculate these values and I wanted to give hints to future generations.
The code is needed as a last check of the successful bond redemption ("bank got the money").

File: src/main/java/com/digitalasset/refapps/bondissuance/util/PartyAllocator.java [hunk 3]
```diff
@@ -7,101 +7,172 @@
...
-  private static String allocateParty(
-      PartyManagementServiceGrpc.PartyManagementServiceBlockingStub stub, String party) {
-    PartyManagementServiceOuterClass.AllocatePartyResponse response =
-        stub.allocateParty(
-            PartyManagementServiceOuterClass.AllocatePartyRequest.newBuilder()
-                .setDisplayName(party)
-                .setPartyIdHint(party)
-                .build());
-    return response.getPartyDetails().getParty();
+  private static void tryToAllocateParty(
```
Comment (tamaskalcza-da) [inline src/main/java/com/digitalasset/refapps/bondissuance/util/PartyAllocator.java]: I think this does too much and we can simplify it by separating some concerns. What I have in mind is something like the following:
- return `Optional<String>`
- in the client code of this method use `getOrElse` to fallback to `alreadyKnonwParties`
- create the `parties` map in `allocate` from the individual parties
Not hundred percent sure about this, however I think it worths to consider.
